qabot/pipeline_maintenance.py

```diff
@@ -213,8 +213,10 @@ def ci_benchmarking(self, repo_name, pr_num, stage_name):
             stage_duration = jl.get_duration_of_ci_pipeline_stage(
                 repo_name, pr_num, stage_name
             )
-            duration_raw = datetime.datetime.fromtimestamp(stage_duration / 1000.0)
-            friendly_duration_format = duration_raw.strftime("%Mm and %Ss")
+            seconds = stage_duration / 1000
```

Comment:
the seconds computation would be wrong in the final result
>>> duration=3600000
>>> seconds = duration/1000
>>> minutes = seconds/60
>>> hours = minutes/60
>>> friendly_duration_format = f"{str(hours % 60)[0:1]}h {str(minutes % 60)[0:2]}m {str(seconds)[0:2]}s"
>>> print(friendly_duration_format)
1h 0.m 36s

This should do the trick:
>>> str(datetime.timedelta(milliseconds=3661010))
'1:01:01.010000'

Or, to get a human friendly format, this:
>>> delta = str(datetime.timedelta(milliseconds=3661010)).split(".")[0].split(":")
>>> print(" ".join([i+j for i,j in zip(delta, ['h', 'm', 's'])]))
1h 01m 01s